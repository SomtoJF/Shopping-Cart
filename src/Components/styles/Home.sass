@use '../../App'

#home-main
    display: flex
    flex-direction: column
    padding-bottom: 100px

    #buttons
        width: 100vw
        height: fit-content
        background-color: App.$primary-color
        display: flex
        justify-content: center
        gap: 10px

        a
            @include App.link($width: 100px)
            

    
#logo
    z-index: 1
    margin: 0
    margin-bottom: -10%
    padding: 0
    width: fit-content
    font-size: 50vw
    font-family: App.$logo-font
    align-self: center
    font-weight: 600

#homepage-image
    background-image: url('https://images.pexels.com/photos/5710990/pexels-photo-5710990.jpeg')
    background-repeat: no-repeat
    background-size: cover
    width: 100vw
    height: 50vh
    align-self: center
    box-sizing: border-box
    display: flex
    align-items: flex-end
    figcaption
        @include App.link
    a:visited
        color: App.$primary-color
        

#intro-text
    font-family: App.$info-text-font
    margin: 15px 0px

@media all and (min-width: 500px) 
    #logo
        font-size: 40vw

@media all and (min-width: 1300px) 
    #logo
        font-size: 30vw
    #homepage-image
        figcaption
            padding-left: 10px


@media all and (max-width: 1000px) 
    #logo
        word-break: normal
        height: 60vh
        margin-bottom: -60%
        align-self: flex-start
