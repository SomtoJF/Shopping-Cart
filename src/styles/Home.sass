@use '../App'

#home-main
    display: flex
    flex-direction: column
    padding-bottom: 100px

    #buttons
        width: 100vw
        height: fit-content
        background-color: App.$primary-color
        display: flex
        justify-content: center
        gap: 10px

        a
            @include App.link($width: 100px)
            

    
#logo
    z-index: 1
    margin: 0
    margin-bottom: -10%
    padding: 0
    width: fit-content
    font-size: 50vw
    font-family: App.$logo-font
    align-self: center
    font-weight: 600
    animation: fadeIn 2s ease 0s 1 normal forwards

@keyframes fadeIn
    0% 
        opacity: 0
        transform: translateX(-50px)

    100% 
        opacity: 1
        transform: translateX(0)
	

#homepage-image
    background-image: url('https://images.pexels.com/photos/12367865/pexels-photo-12367865.jpeg')
    background-repeat: no-repeat
    background-size: cover
    width: 100vw
    height: 50vh
    align-self: center
    box-sizing: border-box
    display: flex
    align-items: flex-end
    figcaption
        @include App.link
    a:visited
        color: App.$primary-color
        

#intro-text
    font-family: App.$info-text-font
    margin: 15px 0px

@media all and (min-width: 500px) 
    #logo
        font-size: 40vw

@media all and (min-width: 1300px) 
    #logo
        font-size: 30vw
    #homepage-image
        figcaption
            padding-left: 10px

@media all and (min-width: 1000px) 
    #homepage-image
        background-image: url(https://images.pexels.com/photos/12095123/pexels-photo-12095123.jpeg)
        background-size: 100%
        background-position: center



@media all and (max-width: 1000px) 
    #logo
        word-break: normal
        height: 60vh
        margin-bottom: -60%
        align-self: flex-start
