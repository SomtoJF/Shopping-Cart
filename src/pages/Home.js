import { Link } from "react-router-dom";
import "../styles/Home.sass";

const Home = () => {
	return (
		<main id="home-main">
			<h1 id="logo">OUT SPOKEN</h1>
			<figure id="homepage-image" alt="image">
				<figcaption>
					<a href="https://www.pexels.com/@anastasiia-chaikovska-206547003/">
						Anastasiia Chaikovska
					</a>
				</figcaption>
			</figure>
			<article id="intro-text">
				<b>AWOKEN | UNBROKEN | OUTSPOKEN</b> - this is what we believe in. Since
				2016, Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec
				tristique ipsum eu blandit pharetra. Praesent a ornare neque. Duis
				laoreet magna id pretium finibus. Vestibulum ut urna volutpat, mollis
				nunc eu, varius augue. Etiam imperdiet, arcu vitae cursus commodo, purus
				enim imperdiet massa, quis faucibus nisi quam vel enim. Vestibulum.
			</article>
			<div id="buttons">
				<Link to="/Shopping-Cart/products">Shop Now</Link>
				<Link to="/Shopping-Cart/cart">Your Cart</Link>
			</div>
		</main>
	);
};

export default Home;
