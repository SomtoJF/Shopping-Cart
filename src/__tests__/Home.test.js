import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom"; // optional
import userEvent from "@testing-library/user-event";
import Home from "../pages/Home";

describe("Home Component", () => {
	it("renders correctly", () => {
		const { container } = render(<Home />, { wrapper: MemoryRouter });
		expect(container).toMatchSnapshot();
	});
});
