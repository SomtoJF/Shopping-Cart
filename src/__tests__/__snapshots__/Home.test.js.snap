// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders correctly 1`] = `
<div>
  <main
    id="home-main"
  >
    <h1
      id="logo"
    >
      OUT SPOKEN
    </h1>
    <figure
      alt="image"
      id="homepage-image"
    >
      <figcaption>
        <a
          href="https://www.pexels.com/@anastasiia-chaikovska-206547003/"
        >
          Anastasiia Chaikovska
        </a>
      </figcaption>
    </figure>
    <article
      id="intro-text"
    >
      <b>
        BE HEARD! BE STRONG! BE PROUD!
      </b>
       - this is what we believe in. Since 2016, Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec tristique ipsum eu blandit pharetra. Praesent a ornare neque. Duis laoreet magna id pretium finibus. Vestibulum ut urna volutpat, mollis nunc eu, varius augue. Etiam imperdiet, arcu vitae cursus commodo, purus enim imperdiet massa, quis faucibus nisi quam vel enim. Vestibulum.
    </article>
    <div
      id="buttons"
    >
      <a
        href="/products"
      >
        Shop Now
      </a>
      <a
        href="/cart"
      >
        Your Cart
      </a>
    </div>
  </main>
</div>
`;
